### PR TITLE
Synapsify combined studies

### PR DESCRIPTION
**Pull request by markdane**
@kdaily Modified the combined studies dataset to read level 3 and 4 files from multiple studies from Synpase, normalize the common signals, add in the staining set specific signals then write out levels 3 and 4 with annotations and provenances.

This commit also include error checking on the signals in the QA_SpotMEP report and adds some uploader files to the archive folder.

### CODE REVIEW

File: Pipeline/PreprocessCombinedStudies.R [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                help="Print extra output"),
+    make_option(c("-k", "--k"), type="integer", default=256,
+                help="Number of factors to use in RUV normalization [default %default]",
+                metavar="number"),
+    make_option(c("-i", "--inputPath"), type="character", default=NULL, metavar="PATH",
+                help="Path to local input data directory or Synapse ID for a File View."),
+    make_option(c("--synapseStore"), type="character", default=NULL, metavar="SYNAPSEID",
+                help="Store output file in Synapse directory (provide Synapse ID of Folder to store).")
+  )
+  parser <- OptionParser(usage = "%prog [options] barcode file", option_list=option_list)
```
Comment (kdaily) [inline Pipeline/PreprocessCombinedStudies.R]: BARCODE OUTPUTFILE

File: Pipeline/PreprocessCombinedStudies.R [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+###Debug
+cl <-list(options=list(verbose=TRUE,
+                       inputPath="syn7494072",
+                       k=256,
+                       synapseStore='syn5713302'),
+          args=c("HMEC122L_SS1","HMEC122L_SS4"))
+####
+#Get the command line arguments and options
+cl <- getCSCommandLineArgs()
```
Comment (kdaily) [inline Pipeline/PreprocessCombinedStudies.R]: getCommandLineArgs

File: Pipeline/PreprocessCombinedStudies.R [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+getCSCommandLineArgs <- function(){
+  option_list <- list(
+    make_option(c("-v", "--verbose"), action="store_true", default=FALSE,
+                help="Print extra output"),
+    make_option(c("-k", "--k"), type="integer", default=256,
+                help="Number of factors to use in RUV normalization [default %default]",
+                metavar="number"),
+    make_option(c("-i", "--inputPath"), type="character", default=NULL, metavar="PATH",
+                help="Path to local input data directory or Synapse ID for a File View."),
```
Comment (kdaily) [inline Pipeline/PreprocessCombinedStudies.R]: can't give local file path - change param name to make it more clear that it works differently from others

File: Pipeline/PreprocessCombinedStudies.R [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+suppressMessages(synapseLogin())
+#Get annotations from the level 3 files
+annotations <- rbindlist(lapply(studyNameList, function(studyName){
+  levelQuery <- sprintf('SELECT id,Segmentation,Preprocess,Study,DataType,Consortia,StainingSet,CellLine,Drug from %s WHERE Study="%s" AND Level="%s"',
+                        cl$options$inputPath, studyName, 3)
+  levelRes <- synTableQuery(levelQuery)
+  levelRes@values
+}))
+#Check if any annotations contain more than one value 
+if(nrow(unique(annotations[,.(Study,Segmentation,Preprocess,DataType,Consortia,Drug)]))>1) stop("Cannot comnbine studies with mixed annotations")
```
Comment (kdaily) [inline Pipeline/PreprocessCombinedStudies.R]: typo

File: Pipeline/PreprocessCombinedStudies.R [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+mepDT <- preprocessLevel4(l3C[,grep("Endpoint|StainingSet|395nm|488nm|555nm|640nm|750nm|1An$|2An$|3An$|LigandSet",colnames(l3C),value=TRUE,invert=TRUE), with=FALSE],seNames=c("DNA2N","SpotCellCount","EdU","MitoTracker","KRT","Lineage","Fibrillarin"))
+
+#Add in the barcodes for each MEP_Drug
+mepDT <- addBarcodes(dt3 = l3C, dt4 = mepDT)
+
+# Add a QA flag for spots with few replicates
+mepDT$QA_LowReplicateCount <- mepDT$Spot_PA_ReplicateCount < 3
+
+message("Writing level 4 file to disk\n")
+write.table(mepDT, paste0(path,"/study/",studyNameSSC,"/Annotated/",studyNameSSC,"_Level4.tsv"), sep = "\t",row.names = FALSE, quote=FALSE)
```
Comment (kdaily) [inline Pipeline/PreprocessCombinedStudies.R]: temp folder

**Review by kdaily — CHANGES_REQUESTED** [review]

File: Pipeline/PreprocessCombinedStudies.R (hunk 3 above)
Comment (markdane) [reply Pipeline/PreprocessCombinedStudies.R]: make_option(c("-i", "--inputSynID"), type="character", default=NULL, metavar="synID",
                help="Synapse ID for a File View with input data."),

**Pull request merged**